### PR TITLE
fix: delete init.cache rather than assign undefined

### PR DESCRIPTION
**Pull request by vicb**
Fix an issue with NextAuth (#206)

Creating a PR so that users can test, but need to come with a proper test

### CODE REVIEW

File: packages/cloudflare/src/cli/build/bundle-server.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ fetch = globalThis.fetch;
 const CustomRequest = class extends globalThis.Request {
   constructor(input, init) {
     if (init) {
-      init.cache = undefined;
+      delete init.cache;
```
Comment (vicb) [inline packages/cloudflare/src/cli/build/bundle-server.ts:97]: Using `Object.defineProperty()` would throw later on l88.

**Review by dario-piotrowicz — APPROVED** [review]
Looks good to me 👍 

The first thing that came to mind when seeing the change is that a comment could help clarify why we're not setting the value to `undefined` but not having the comment is not a huge deal, especially since this seems to be a bit of a specific case

Besides (and related to) the above one worry I have here is that I feel that there is nothing to prevent this code to be changed in the future reintroducing the issue, I think it would be very good if we could have some e2e testing here, but I also understand that this might be a bit overkill and not worth it right now, so I am fine skipping that too

File: packages/cloudflare/src/cli/build/bundle-server.ts (hunk 1 above)
Comment (dario-piotrowicz) [reply packages/cloudflare/src/cli/build/bundle-server.ts:97]: I find this and l88 quite cryptic 🥲, hopefully this code will be improved/made clearer at some point

**Comment by vicb** [comment]
We should definitely find a way to test that (as mentioned in the description).
I'll create a separate issue for that.
Merging as the reporter of #206 saw an improvement.

**Pull request merged**